V3 AI Dispatcher

dias-bakhtiyarov opened the pull request:
**feat(proto): add AIDispatcherV3 service definition**
- Introduce `proto_schema/v3_dispatcher.proto` for AI Dispatcher V3.
- Add `AssignmentRequest` and `AssignmentResponse` messages.
- Define `MakeAssignment` RPC endpoint.
- Import necessary dependencies (`google/protobuf`, `job.proto`, `route.proto`).

Commit:
**fix(proto): make `depot_id` and `expected_location_id` optional in `DriverOnShift`**

- Updated `depot_id` and `expected_location_id` fields to optional in `DriverOnShift` message definition.

**feat(proto): add new route event `ROUTE_EVENT_CREATE_ROUTE_BY_AI`**

- Introduced `ROUTE_EVENT_CREATE_ROUTE_BY_AI` event in `route.proto`.
- Removed unused imports from `v3_dispatcher.proto`.

Kuandykzx commented on driver_on_shift.proto:
```diff
@@ -57,25 +55,25 @@ message DriverOnShift {
   optional google.protobuf.Timestamp tw_start_at = 15;
   optional google.protobuf.Timestamp tw_end_at = 16;
   optional uint64 capacity = 17;
+  required uint64 depot_id = 18;
```
they need to be optional, will break current stuff

Kuandykzx commented on driver_on_shift.proto:
```diff
@@ -57,25 +55,25 @@ message DriverOnShift {
   optional google.protobuf.Timestamp tw_end_at = 16;
   optional uint64 capacity = 17;
+  required uint64 depot_id = 18;
+  required uint64 expected_location_id = 19;
```
expected at the end?

dias-bakhtiyarov replied:
Expected location at the nearest/next availability time (tw_start_at).

Kuandykzx commented on v3_dispatcher.proto:
```diff
@@ -0,0 +1,23 @@
+message AssignmentRequest {
+  required uint64 service_provider_id = 1;
+  required string event = 2;
+  required uint64 target_entity_id = 3;
```
what does it mean? maybe more concrete ( route_id, job_id, shift_id etc)

dias-bakhtiyarov replied:
It's either job_id or shift_id depending on event (idle driver or new job).

Commit:
**feat(proto): add `depot_id` and `visible_to_ai` to `Truck` message definitions**

- Added `depot_id` (required) and `visible_to_ai` (optional) fields to `Truck`, `TruckCreate`, and `TruckUpdate` message definitions.

**feat(proto): add `visible_to_ai` to `Job` message definitions**

- Added `visible_to_ai` field to `Job`, `JobCreate`, and `JobUpdate` message definitions.

**feat(proto): add `priority` to `Job` and `visible_to_ai` to `DriverOnShift`**

- Introduced `priority` field to `Job`, `JobCreate`, and `JobUpdate` message definitions.
- Added `visible_to_ai` field to `DriverOnShift`, `DriverOnShiftCreate`, and `DriverOnShiftUpdate` message definitions.
